src/foam/nanos/crunch/CapabilityIsVisible.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'CapabilityIsVisible',
+  extends: 'foam.mlang.predicate.AbstractPredicate',
+  implements: [ 'foam.core.Serializable' ],
+
+  javaImports: [
+    'foam.core.FObject',
+    'foam.core.XLocator',
+    'foam.core.X',
+    'foam.nanos.crunch.Capability'
+  ],
+
+  methods: [
+    {
+      name: 'f',
+      code: () => { return true; },
+      javaCode: `
+        if ( ! ( obj instanceof Capability) ) return false;
+        X x = XLocator.get();
+        Capability capability = (Capability) obj;
+        return capability.getVisible() ? capability.getVisibilityPredicate() != null 
```

Comment:
Why not just replace the "Visible" property with the visibilityPredicate property. 
Then the predicate check is always simple. EQ(visible, true); or PropertyEQValue(visible,true)
Or if its because you want the ui checkbox just do: 
PropertyEQValue(visible,true) && PropertyEQValue(visibilityPredicate,true). 
Then you dont need this custom predicate at all

Reply:
@yij793 can you respond to @nanoMichal comments, please.

Reply:
@jlhughes I did with slack, this isn't works on client side

Reply:
I still find it really odd that you need this just to check a boolean.  Is it a bug or missing support making it not possible?

Reply:
@jlhughes In client side, PropertyEQValue's prop does not accept predicate, same as EQ.